packages/svelte/src/internal/client/runtime.js

```diff
@@ -895,15 +895,17 @@ function mark_signal_consumers(signal, to_status, force_schedule) {
 		for (i = 0; i < length; i++) {
 			const consumer = consumers[i];
 			const flags = consumer.flags;
+			const unowned = (flags & UNOWNED) !== 0;
+			const dirty = (flags & DIRTY) !== 0;
 			if (
-				(flags & DIRTY) !== 0 ||
+				(dirty && !unowned) ||
 				(!runes && consumer === current_effect) ||
 				(!force_schedule && consumer === current_effect)
 			) {
 				continue;
 			}
 			set_signal_status(consumer, to_status);
-			if ((flags & CLEAN) !== 0) {
+			if ((flags & CLEAN) !== 0 || (dirty && unowned)) {
```

Comment:
I don't fully understand how this resolves the issue - could add a small comment? Issues like these feel like good opportunities to add more explanatory comments to the runtime - this stuff is super tricky, so I think every little bit helps 😃